src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable react/sort-comp, react/button-has-type */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { selectableAssignmentLabels } from 'data/selectors/filters';
+import * as gradesActions from 'data/actions/grades';
+
+import SelectGroup from '../SelectGroup';
+
+export class AssignmentTypeFilter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    const assignmentType = event.target.value;
+    this.props.filterAssignmentType(assignmentType);
+    this.props.updateQueryParams({ assignmentType });
+  }
+
+  get options() {
+    const mapper = (entry) => (
+      <option key={entry} value={entry}>{entry}</option>
+    );
+    return [
+      <option key="0" value="">All</option>,
+      ...this.props.assignmentTypes.map(mapper),
+    ];
+  }
+
+  render() {
+    return (
+      <div className="student-filters">
+        <SelectGroup
+          id="assignment-types"
+          label="Assignment Types"
+          value={this.props.selectedAssignmentType}
+          onChange={this.handleChange}
+          disabled={this.props.assignmentFilterOptions.length === 0}
```

Comment:
I know this is a refactor but this seems like it's a bug .... should the Assignment Types dropdown really be disabled when there are no selectable assignments for that Assignment Type? That doesn't make sense to me.

Reply:
There are... a few logic inconsistencies with the current implementation that I feel should be reviewed. (some "disabled" filters can also still be saved in the filterBadges...